src/ncp/ncp_base.cpp

```diff
@@ -348,13 +411,20 @@ void NcpBase::HandleReceive(const uint8_t *aBuf, uint16_t aBufLength)
 
     mRxSpinelFrameCounter++;
 
-    // We only support IID zero for now.
-    if (SPINEL_HEADER_GET_IID(header) != 0)
+    mCurCommandIid = SPINEL_HEADER_GET_IID(header);
+
+#if OPENTHREAD_CONFIG_MULTIPAN_RCP_ENABLE
+    if (mCurCommandIid > SPINEL_HEADER_IID_MAX)
```

Comment:
I am wondering if we should also add this condition `(mCurCommandIid > SPINEL_HEADER_IID_MAX) ||(mCurCommandIid > OPENTHREAD_CONFIG_MULTIPLE_STATIC_INSTANCE_ENABLE && mCurCommandIid != OPENTHREAD_SPINEL_CONFIG_BROADCAST_IID)`. The reason I am thinking is, let's say `OPENTHREAD_SPINEL_CONFIG_BROADCAST_IID` is set to 0 and `OPENTHREAD_CONFIG_MULTIPLE_STATIC_INSTANCE_ENABLE` set to 1, then we cannot have a host with iid = 2. There has to be a host that uses iid = 1; otherwise, `IidToInstance()` will return nullptr. This condition is mainly to catch this misconfiguration of host. Does this make sense? Or do we have a better way to catch such a scenario?

OR

Other option is to return `SPINEL_STATUS_INVALID_INTERFACE`  if `mInstance == nullptr` after line 426.

Reply:
Used the second approach as I think in the future someone may want to have instances dynamically allocated on the RCP. In that case using ...STATIC_INSTANCE.. macros will not work.